br/pkg/aws/ebs.go

```diff
@@ -281,6 +281,116 @@ func (e *EC2Session) DeleteSnapshots(snapIDMap map[string]string) {
 	log.Info("delete snapshot end", zap.Int("need-to-del", len(snapIDMap)), zap.Int32("deleted", deletedCnt.Load()))
 }
 
+// EnableDataFSR enables FSR for data volume snapshots
+func (e *EC2Session) EnableDataFSR(meta *config.EBSBasedBRMeta, targetAZ string) ([]*string, error) {
+	sourceSnapshotIDs, availableZones := fetchTargetSnapshots(meta, targetAZ)
+
+	log.Info("Start enable FSR for", zap.Int("snapshot number", len(sourceSnapshotIDs)), zap.Any("Snapshots", sourceSnapshotIDs), zap.String("available zone", targetAZ))
+
+	resp, err := e.ec2.EnableFastSnapshotRestores(&ec2.EnableFastSnapshotRestoresInput{
+		AvailabilityZones: availableZones,
+		SourceSnapshotIds: sourceSnapshotIDs,
+	})
+
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	if len(resp.Unsuccessful) > 0 {
+		log.Warn("not all snapshots enabled FSR")
+		return nil, errors.Errorf("Some snapshot fails to enable FSR, such as %s, error code is %v", *resp.Unsuccessful[0].SnapshotId, resp.Unsuccessful[0].FastSnapshotRestoreStateErrors)
+	}
+
+	log.Info("Enable FSR issued")
+
+	return sourceSnapshotIDs, nil
+}
+
+// WaitDataFSREnabled waits FSR for data volume snapshots are all enabled
+func (e *EC2Session) WaitDataFSREnabled(snapShotIDs []*string, targetAZ string, progress glue.Progress) error {
+	pendingSnapshots := make([]*string, 0, len(snapShotIDs))
+
+	pendingSnapshots = append(pendingSnapshots, snapShotIDs...)
+
+	log.Info("starts check fsr pending snapshots", zap.Any("snapshots", pendingSnapshots))
+	for {
+		if len(pendingSnapshots) == 0 {
+			log.Info("all snapshots fsr enablement is finished.")
+			return nil
+		}
+
+		// check pending snapshots every 1 minute
+		time.Sleep(1 * time.Minute)
+		log.Info("check snapshots not fsr enabled", zap.Int("count", len(pendingSnapshots)))
+		input := &ec2.DescribeFastSnapshotRestoresInput{
+			Filters: []*ec2.Filter{
+				{
```

Comment:
we should add the `snapshot-id` filter. We should only watch the fsr we created.